python/google/protobuf/descriptor.py

```diff
@@ -874,7 +874,7 @@ def __new__(cls, name, package, options=None,
       # FileDescriptor() is called from various places, not only from generated
       # files, to register dynamic proto files and messages.
       # pylint: disable=g-explicit-bool-comparison
-      if serialized_pb == '':
+      if serialized_pb == b'':
```

Comment:
This is used to distinguish the linked Cpp generated code usage, which serialized_pb is not really needed. If we change this, it requires to change the generator as well:
https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/compiler/python/python_generator.cc#L470

Also change this will cause compatibility issue: old generated code may not work with new runtime.

Reply:
Thanks for the feedback @anandolee !  Updated the generator.

The change introduced in `3.11` already caused compatibility issues - before 3.11 this branch would be take on empty bytes, now it won't:

```
In [1]: if '' == b'':
   ...:     print('Empty Bytes Are Empty Strings')
   ...: if not b'':
   ...:     print('Empty Bytes Are False-y')
   ...:
Empty Bytes Are False-y
```

Updated to check the length as this is the fastest way to determine if it's either `b''` or `''` without hitting a `BytesWarning`.

```python
In [1]: x = b''

In [2]: %timeit len(x) == 0
72.1 ns ± 0.787 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %timeit (isinstance(x, str) and x == '') or (isinstance(x, bytes) and x == b'')
233 ns ± 8.01 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```